test/helpers/kubectl.go

```diff
@@ -1130,3 +1130,102 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 
 	return res.Output().String(), nil
 }
+
+// CiliumPreFlightCheck specify that it checks that various subsystems within
+// Cilium are in a good state. If one of the multiple preflight fails it'll
+// return an error.
+func (kub *Kubectl) CiliumPreFlightCheck() error {
+	// Doing this withTimeout because the Status can be ready, but the other
+	// nodes cannot be show up yet, and the cilium-health can fail as a false positive.
+	var err error
+	body := func() bool {
+		err = kub.CiliumControllersPreFlightCheck()
+		if err != nil {
+			return false
+		}
+		err = kub.CiliumHealthPreFlightCheck()
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	timeoutErr := WithTimeout(body, "PreflightCheck failed", &TimeoutConfig{Timeout: HelperTimeout})
```

Comment:
Hi, this function return err, that it's the same variable that it's used on the fail. 
```
var err error
+	body := func() bool {
+		err = kub.CiliumControllersPreFlightCheck()
....
timeoutErr := WithTimeout(body, "PreflightCheck failed", &TimeoutConfig{Timeout: HelperTimeout})
if timeoutErr != nil {
+		return err
+	}
+	return nil
```

So it'll return the preflight error.

Reply:
I'm talking about this:
```
        for {
                select {
                case <-ticker.C:
                        if body() {
                                return nil
                        }
                case <-done:
                        return fmt.Errorf("Timeout reached: %s", msg)
                }
        }
```

Reply:
done

Reply:
@tgraf do you mean you want `WithTimeout` to do the bookkeeping of the error from `body`, instead of the caller?
@eloycoto this might be worth changing. `body` can still do what it does here, managing the `err` object directly, but a simpler situation might just want to get the last error without writing the code for it (e.g. an `if err := WithTimeout; err != nil {`.
I'm not sure how best to indicate the timeout case. Wrapping the error from `body` is probably fine, but multiple return values might be more clear. My fear is that wrapping the error loses the type and we sometimes want to check that the error is EOF or of a specific class.